gaps/gap-4_comp_manifest/gap-4_comp_manifest.md

```diff
@@ -0,0 +1,271 @@
+computations they may execute on a Provider's machine.
+
+## Motivation
+
+The need for a Computation Manifest originates from the upcoming Internet
+access feature, where each outbound request is routed via Provider's host
+operating system. In order to prevent unwanted / illegal behavior on Provider's
+behalf, she needs to remain in control of actions that Requestors can execute
+on her machine.
+
+With Computation Manifests, Requestors constrain themselves to
+a certain set of allowed actions, to be negotiated with and approved by a
+Provider. Requestors' actions will be verified against the Manifest during
+computation.
+
+## Specification
+
+Computation Manifest can take one of the following forms:
+
+- an embedded JSON / YAML object within an external configuration file,
+- `golem.srv.comp.manifest` namespace properties, an extension to
+  Golem Computation Resource Standards.
+
+The Computation Manifest is expected be included in a Computation Payload
+Manifest ([GAP](https://github.com/golemfactory/golem-architecture/pull/28))
+file, where its hash is co-signed with Payload hashes. If a Computation
+Manifest is included in the Demand (and thus, Agreement), it's role is to
+derive the properties from and override the initially sourced Manifest
+### GCRS `golem.srv.comp.manifest` namespace
+
+Intended for overriding an external Manifest but allows defining a new Manifest
+from the ground up using properties only.
+
+1. `golem.srv.comp.manifest.version : String`
+
+    Specifies a version (Semantic Versioning 2.0 specification) of the manifest,
+    **defaults** to "0.1.0"
+
+2. `golem.srv.comp.manifest.script` (sub-namespace)
+
+    Defines a set of allowed ExeScript commands and applies constraints to
+    their arguments.
+
+
+3. `golem.srv.comp.manifest.script.commands : List[String]`
+
+    Specifies a curated list of commands in form of:
+
+    - UTF-8 encoded strings
+
+      No command context or matching mode need to be specified.
+
+      E.g. `["run /bin/cat /etc/motd", "run /bin/date -R"]`
+
+    - UTF-8 encoded JSON strings
+
+      Command context (e.g. `env`) or argument matching mode need to be
+      specified for a command.
+
+      E.g. `["{\"run\": { \"args\": \"/bin/date -R\", \"env\": { \"MYVAR\": \"42\", \"match\": \"strict\" }}}"]`
+
+    - mix of both
+
+    `"deploy"`, `"start"` and `"terminate'` commands are always allowed.
+    These values become the **default** if no `manifest.script.command` property
+    has been set in the Demand, but the `manifest` namespace is present.
+
+4. `golem.srv.comp.manifest.script.match : String`
+
+    Selects a default way of comparing command arguments stated in the manifest
+    and the ones received in the ExeScript, unless stated otherwise in a
+    command JSON object.
+
+    The `match` property could be one of:
+
+      - `"strict"`: byte-to-byte argument equality (**default**)
+      - `"regex"`: treat arguments as regular expressions (UTF-8 Unicode mode)
```

Comment:
Can we please specify the regex standard to be used here, or link to eg. regex syntax spec that will be used?